File: services/heartbeat_line.go
package services

import "fmt"

// formatHeartbeatLine builds the one-line "heartbeat processed" confirmation shown in a
// Go-scheduled agent's dashboard console (carried there by the Node stdout/stderr→goLog
// forwarder in orchestrator.js). Mirrors the Node agent/heartbeat-line.js format so the
// indicator reads consistently across LLM-beat and Go-scheduled agents.
func formatHeartbeatLine(agent, summary, etTime string) string {
	who := ""
	if agent != "" {
		who = agent + " "
	}
	ctx := ""
	if etTime != "" {
		ctx = " · " + etTime + " ET"
	}
	return fmt.Sprintf("%s✓ heartbeat processed — %s%s", who, summary, ctx)
}

File: agent/heartbeat-line.js
// One-line "heartbeat processed" confirmation for an agent's dashboard console pane.
// Pure formatter — the wiring (harness agent_log emit for Node agents; Go scheduler log line
// for Go-scheduled agents) calls this and routes the string through the existing console plumbing.
// Printed only on a PROCESSED beat (skipped/preflight beats already emit their own line).
export function formatHeartbeatLine({ agent, phase, etTime, toolCalls = 0, trades = 0 } = {}) {
  const who = agent ? `${agent} ` : '';
  const tc = `${toolCalls} tool call${toolCalls === 1 ? '' : 's'}`;
  const tr = `${trades} trade${trades === 1 ? '' : 's'}`;
  const ctx = [phase, etTime ? `${etTime} ET` : null].filter(Boolean).join(' ');
  const suffix = ctx ? ` · ${ctx}` : '';
  return `${who}✓ heartbeat processed — ${tc}, ${tr}${suffix}`;
}
